### PR TITLE
nrf/boards/common.ld: Avoid overflowing the .text region.

### PR DESCRIPTION
**Pull request by aykevl**
Similar commit to this one:
https://github.com/tralamazza/micropython/commit/6e56e6269f467e59316b5e4cb04ea37ab6a0dfe3

When .text + .data oveflow available flash, the linker may not show an
error. This change makes sure .data is included in the size calculation.

For some background, see the excellent [Red Had documentation](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/4/html/Using_ld_the_GNU_Linker/sections.html#OUTPUT-SECTION-LMA), in particular this one:

> The linker will normally set the LMA equal to the VMA. You can change that by using the `AT` keyword. The expression lma that follows the `AT` keyword specifies the load address of the section. Alternatively, with `AT>lma_region` expression, you may specify a memory region for the section's load address.

In other words, using just the load address (`AT(load_address)`) we can set the initial pointer from where to write (with no limit), while the `AT>lma_region` actually places the data in a section limiting it's size. At least, that's how I understand it and from what I remember from DFU development this fixed an issue where the linker would silently succeed but would write just outside of the specified memory area causing weird issues. After this fix, the linker gave an error, as it should.

### CODE REVIEW

**Comment by glennrub** [comment]
Looks like a good improvement.

Did you see any board targets in current master that struggled fitting into .text at the moment, that should be looked at?

**Comment by aykevl** [comment]
I haven't looked that closely. If they fail after this patch, they were doing something invalid anyway.

Note that .data is usually very small, a few hundred bytes max and can be optimized to 4-8 bytes (current nrfx HAL port), so this issue will only trigger in this very narrow range, and then only if the flash filesystem is used.

**Comment by glennrub** [comment]
I've tested to compile all targets, with and without SD enabled. They all compile as expected.

This also triggered a unseen compilation error, and PR #154. This bug/fix is unrelated.
